### PR TITLE
Update/0.3.25

### PR DESCRIPTION
**Pull request by ELundby45**
# Jaxlib 0.3.25

upstream: https://github.com/google/jax/tree/main/jaxlib
jira: https://anaconda.atlassian.net/browse/PKG-771
`setup.py`: https://github.com/google/jax/blob/main/jaxlib/setup.py

### Changes:
- Update version and SHA
- Remove cross-compilation logic
- OSX: Use SDK10.14
- Update dependencies and pinnings
- Adds additional tests 
- Update about section

### Notes:
- Currently lacking GPU/Cuda support. MCU only
- `ppc64le` and `s390x` are skipped due to lacking `bazel`
- `win64` skipped due to lack of upstream support

### CODE REVIEW

**Review by varlackc — COMMENTED** [review]
The package `opt_einsum` seems too be a requirement for this package:
https://github.com/google/jax/blob/jaxlib-v0.3.25/setup.py#L68

**Comment by ELundby45** [comment]
> 
> The package `opt_einsum` seems too be a requirement for this package:
> https://github.com/google/jax/blob/jaxlib-v0.3.25/setup.py#L68
> 

I believe that is for jax and not jaxlib. 

https://github.com/google/jax/blob/main/jaxlib/setup.py#L49

File: recipe/meta.yaml [hunk 1]
```diff
@@ -7,56 +7,63 @@ package:
...
+    - python
+    # https://github.com/google/jax/blob/jaxlib-v0.3.25/build/build.py#L197
+    - bazel >=5.1.1
+    - git
   host:
     - python
     - pip
-    - numpy
-    - scipy
     - wheel
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:27]: `setuptools` is missing in `host`, https://github.com/google/jax/blob/8788a9438f7c7e00ccc8d60f7cb320177e3d9f80/jaxlib/setup.py#L15

Comment (ELundby45) [reply recipe/meta.yaml:27]: Thank you! I've added that.

**Review by skupr-anaconda — APPROVED** [review]

File: recipe/meta.yaml [hunk 2]
```diff
@@ -7,56 +7,64 @@ package:
...
+    - setuptools
+    - numpy 1.21.*  # [py>=310 or (osx and arm64)]
+    - numpy 1.20.*  # [py<310 and not (osx and arm64)]
   run:
-    - python
     - absl-py
```
[the remark this answers is not in the document]
Comment (ELundby45) [reply recipe/meta.yaml]: Thanks! I removed both.

**Comment by aniedzielaAnaconda** [comment]
Did a pair review with @ELundby45, LGTM, approved.

**Review by aniedzielaAnaconda — APPROVED** [review]

**Pull request merged**